packages/terra-application-navigation/src/ApplicationNavigation.jsx

```diff
@@ -328,6 +320,16 @@ const ApplicationNavigation = ({
       closeMenuCallbackRef.current = undefined;
     }
   });
+  
+  useLayoutEffect(() => {
+    if (activeNavigationItemKey !== renderedNavItemKeyRef.current) {
+      setTimeout(() => {
+        focusMainContent();
+      }, 0);
```

Comment:
The timeout is necessary due to the AbstractModal's similar focus logic: https://github.com/cerner/terra-framework/blob/master/packages/terra-abstract-modal/src/inertHelpers.js#L56

Without the timeout, this executes too quickly.

Reply:
Add a comment to the code, I could see this getting 'cleaned up'.